torch/_functorch/aot_autograd.py

```diff
@@ -745,7 +747,10 @@ def from_fun(t):
     @wraps(f)
     def inner(*flat_args):
         # This function is meant to be run with the forward, which expects a flat list of tensor/symint/other args.
-        assert all(isinstance(a, KNOWN_TYPES) for a in flat_args)
+        def valid_input(a):
+            # Insanely sus, talk to folks about this
+            return isinstance(a, KNOWN_TYPES) or callable(flat_args[-1])
```

Comment:
this is related to how hooks are getting passed in right? Can you explain how the current setup interoperates with compiled autograd? I didn't see it in the PR description